allow tasks to run in checkmode

benformosa opened the pull request:
Running a task with async fails in check mode

benformosa committed:
allow tasks to run in checkmode

benformosa committed:
fix typo

benformosa committed:
run non-changing task in checkmode

jamescassell reviewed (changes requested):
Thanks. Just want to make sure it's right.

benformosa committed:
fix another typo

jamescassell committed:
use "no" vs "false" to be consistent

If we ever switch to YAML 1.2, we'll switch everything to "false" but until then...

jamescassell reviewed (approved):
Thanks
